src/Symfony/Component/Config/Resource/GlobResource.php

```diff
@@ -83,12 +83,12 @@ public function serialize()
             $this->hash = $this->computeHash();
         }
 
-        return serialize(array($this->prefix, $this->pattern, $this->recursive, $this->hash));
+        return serialize(array($this->prefix, $this->pattern, $this->recursive, $this->hash, $this->forExclusion, $this->excludedPrefixes));
     }
 
     public function unserialize($serialized)
     {
-        list($this->prefix, $this->pattern, $this->recursive, $this->hash) = unserialize($serialized);
+        list($this->prefix, $this->pattern, $this->recursive, $this->hash, $this->forExclusion, $this->excludedPrefixes) = unserialize($serialized);
```

Comment:
should be `list($this->prefix, $this->pattern, $this->recursive, $this->hash, $this->forExclusion, $this->excludedPrefixes) = unserialize($serialized) + array(4 => false, array());` for BC with existing serialized strings.

Reply:
Should we handle BC for this, since the original change was introduced in 4.2 ?

Reply:
Yes: the BC applies to ppl moving from 4.1 to 4.2. When clearing the cache after upgrading to 4.2, used to be 4.1 apps will failif we don't do this.

Reply:
Yes sure, my bad.